app/src/test/java/com/codesoom/assignment/infra/InMemoryUserRepository.java

```diff
@@ -0,0 +1,54 @@
+package com.codesoom.assignment.infra;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class InMemoryUserRepository implements UserRepository {
```

Comment:
테스트 전용의 클래스는 가끔 시그니처만 보고 구별할 수가 없어서 좀 헷갈릴 때도 있죠. 이런 방법도 생각해볼 수 있습니다.

`TestOnly`라는 이름의 아무 내용 없는 interface를 하나 만들고 이렇게 사용하는 거죠.

```suggestion
public class InMemoryUserRepository implements UserRepository, TestOnly {
```